test/acme/dns/fixture.go

```diff
@@ -82,25 +85,41 @@ func (f *fixture) setup(t *testing.T) func() {
 	f.setupLock.Lock()
 	defer f.setupLock.Unlock()
 
-	if err := validate(f); err != nil {
-		t.Fatalf("error validating test fixture configuration: %v", err)
-	}
+	f.setupRC++
 
-	env, stopFunc := apiserver.RunBareControlPlane(t)
-	f.environment = env
+	// Only run the setup, if there is no instance running already.
+	if f.setupRC == 1 {
+		if err := validate(f); err != nil {
+			t.Fatalf("error validating test fixture configuration: %v", err)
+		}
 
-	cl, err := kubernetes.NewForConfig(env.Config)
-	if err != nil {
-		t.Fatal(err)
-	}
-	f.clientset = cl
+		env, stopFunc := apiserver.RunBareControlPlane(t)
+		f.environment = env
 
-	stopCh := make(chan struct{})
-	f.testSolver.Initialize(env.Config, stopCh)
+		cl, err := kubernetes.NewForConfig(env.Config)
+		if err != nil {
+			t.Fatal(err)
+		}
+		f.clientset = cl
+
+		stopCh := make(chan struct{})
+		f.testSolver.Initialize(env.Config, stopCh)
+
+		f.setupStopFunc = func() {
+			close(stopCh)
+			stopFunc()
+		}
+	}
 
 	return func() {
-		close(stopCh)
-		stopFunc()
+		f.setupLock.Lock()
+		defer f.setupLock.Unlock()
+
+		f.setupRC--
```

Comment:
Hm, instead of reference counting here, what about if we made the `RunConformance` method call something like:

```
f.runBasicNoSetup(t)
f.runExtendedNoSetup(t)
```

and then have `f.RunBasic` call `setup` _and then_ `runBasicNoSetup(t)` etc?

This should keep existing code written against this package working if it was working before (they'll still continue to call `setup`), and anyone calling `RunConformance` alone will then work too

Reply:
@munnerz what do you think?